src/plugins/flow.js

```diff
@@ -21,6 +21,46 @@ pp.flowParseTypeInitialiser = function (tok, allowLeadingPipeOrAnd) {
   return type;
 };
 
+pp.flowParsePredicate = function() {
+  let node = this.startNode();
+  let moduloLoc = this.state.startLoc;
+  let moduloPos = this.state.start;
+  this.expect(tt.modulo);
+  let checksLoc = this.state.startLoc;
+  this.expectContextual("checks");
+  // Force '%' and 'checks' to be adjacent
+  if (moduloLoc.line !== checksLoc.line || moduloLoc.column !== checksLoc.column - 1) {
+    this.unexpected(moduloPos);
+  }
+  if (this.match(tt.parenL)) {
+    this.next();
```

Comment:
Instead of 
```js
if (this.match(tt.parenL)) {
    this.next();
```
you can do 
```js
if (this.eat(tt.parenL)) {
```

Which is doing the exact same thing.